Fri Mar 12 14:07:29 EET 2010  Sergei Trofimovich <slyfox@community.haskell.org>
  * fixup index creation on 64bit-BE platforms (tested on ppc64)
  
  Fixing following test failures:
  ~/devel/hashed-storage $ cabal test
  Storage.Hashed.Tree:
  diffTrees: [Failed]
  ERROR: user error (Descriptor too short in peekItem!)
  Storage.Hashed.Index:
  index listing: [Failed]
  ERROR: user error (Descriptor too short in peekItem!)
  index content: [Failed]
  ERROR: user error (Descriptor too short in peekItem!)
  
         Test Cases  Total
    Passed  0           0
    Failed  3           3
    Total   3           3
diff -rN -u old-hashed-storage/Storage/Hashed/Index.hs new-hashed-storage/Storage/Hashed/Index.hs
--- old-hashed-storage/Storage/Hashed/Index.hs	2010-03-12 14:16:07.323800639 +0200
+++ new-hashed-storage/Storage/Hashed/Index.hs	2010-03-12 14:16:07.342799479 +0200
@@ -152,7 +152,7 @@
         (dsc_fp, dsc_start, dsc_len) = toForeignPtr dsc
     withForeignPtr fp $ \p ->
         withForeignPtr dsc_fp $ \dsc_p ->
-            do pokeByteOff p (off + off_dsclen) dsc_len
+            do pokeByteOff p (off + off_dsclen) (xlate32 $ fromIntegral dsc_len :: Int32)
                memcpy (plusPtr p $ off + off_dsc)
                       (plusPtr dsc_p dsc_start)
                       (fromIntegral dsc_len)
